dbms/src/main/java/org/polypheny/db/transaction/EntityAccessMap.java

```diff
@@ -221,33 +227,31 @@ public Mode getEntityAccessMode( @NonNull EntityAccessMap.EntityIdentifier entit
      * @param table table of interest
      * @return qualified name
      */
-    public EntityIdentifier getQualifiedName( AlgOptTable table, long partitionId ) {
-        if ( !(table instanceof AlgOptTableImpl) ) {
-            throw new RuntimeException( "Unexpected table type: " + table.getClass() );
-        }
-        if ( !(table.getTable() instanceof LogicalTable) ) {
-            throw new RuntimeException( "Unexpected table type: " + table.getTable().getClass() );
-        }
-        return new EntityIdentifier( table.getTable().getTableId(), partitionId, NamespaceLevel.ENTITY_LEVEL );
+    public EntityIdentifier getQualifiedName( Entity table, long partitionId ) {
+        return new EntityIdentifier( table.id, partitionId, NamespaceLevel.ENTITY_LEVEL );
     }
 
 
     /**
      * Visitor that finds all tables in a tree.
      */
-    private class TableRelVisitor extends AlgVisitor {
+    private class TableAlgVisitor extends AlgVisitor {
```

Comment:
This inner class is a bit of a mess. Its naming implies it is relational; however, it is model agnostic. But the visit method we benefit from some cleanup, e.g., by extracting the relational part into a method similar to the doc and graph part.